remediation/code-review/01-countEvens.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Given an array of integers, returns the count of even integers in the array.
+ *
+ * See https://simple.wikipedia.org/wiki/Even_number
+ *
+ * @example
+ * countEvens([1, 2, 3, 4, 5]); // => 2
+ * countEvens([10, 10, 10]); // => 3
+ * countEvens([1, 1, 1, 2]); // => 2
+ *
+ * @param {number[]} array - An array of numbers
+ * @returns {number} The count of even integers in the array
+ */
+function countEvens(array) {
+  for(i=0;i<array.length;i++)
+  {
+    if(array[i]%2 != 0)
+    {
+      array.splice(i,1)
```

Comment:
Removing items that don't fit the condition we want from the list and then counting the remaining items is a good idea, but modifying an array you are currently iterating over has some unintended consequences. Each time you remove an item from the array, the loop will iterate one less time because `array.length` will become smaller. This means it it possible to get an input that will result in this function not even checking every item in the list. How could it be accurate if that is the case? For examply, try adding the test case `[3, 3, 4]` to your sanity checks.

Can you think of a way of keeping count without modifying the array?